lib/core/jmanagedbackends.c

```diff
@@ -0,0 +1,832 @@
+/**
+ * \file JManagedBackends.c
+ * \ingroup backend
+ *
+ * \copyright
+ * JULEA - Flexible storage framework
+ * Copyright (C) 2022 Julian Benda
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+#include <bits/types/struct_timeval.h>
+#include <core/jmanagedbackends.h>
+#include <core/jbackend.h>
+#include <core/jlist.h>
+#include <core/jlist-iterator.h>
+#include <core/jsemantics.h>
+#include <core/jtrace.h>
+#include <core/jconfiguration.h>
+
+#include <julea-config.h>
+
+#include <glib.h>
+
+#include <string.h>
+#include <sys/select.h>
+#include <sys/time.h>
+
+#define EXE(cmd, ...) \
```

Comment:
I would like to have error checking macros on a global level instead of per module. It's probably better to move this one?